core/src/main/java/tc/oc/pgm/modules/StatsMatchModule.java

```diff
@@ -0,0 +1,250 @@
+
+    victimStats.onDeath();
+
+    sendLongHotbarMessage(victim, victimStats.getBasicStatsMessage());
+
+    if (murderer != null
+        && PlayerRelation.get(victim.getParticipantState(), murderer) != PlayerRelation.ALLY
+        && PlayerRelation.get(victim.getParticipantState(), murderer) != PlayerRelation.SELF) {
+      UUID murdererUUID = murderer.getId();
+      PlayerStats murdererStats = allPlayerStats.get(murdererUUID);
+
+      if (hasNoStats(murdererUUID)) murdererStats = putNewPlayer(murdererUUID);
+
+      if (event.getDamageInfo() instanceof ProjectileInfo) {
+        murdererStats.setLongestBowKill(
+            victim
+                .getState()
+                .getLocation()
+                .distance(((ProjectileInfo) event.getDamageInfo()).getOrigin()));
+      }
+
+      murdererStats.onMurder();
+
+      sendLongHotbarMessage(murderer, murdererStats.getBasicStatsMessage());
+    }
+  }
+
+  @EventHandler
+  public void onMatchEnd(MatchFinishEvent event) {
+    Map<UUID, Integer> allKills = new HashMap<>();
+    Map<UUID, Integer> allKillstreaks = new HashMap<>();
+    Map<UUID, Integer> allDeaths = new HashMap<>();
+    Map<UUID, Integer> allBowshots = new HashMap<>();
+
+    for (Map.Entry<UUID, PlayerStats> mapEntry : allPlayerStats.entrySet()) {
+      UUID playerUUID = mapEntry.getKey();
+      PlayerStats playerStats = mapEntry.getValue();
+
+      if (hasNoStats(playerUUID)) playerStats = putNewPlayer(playerUUID);
+
+      allKills.put(playerUUID, playerStats.kills);
+      allKillstreaks.put(playerUUID, playerStats.killstreakMax);
+      allDeaths.put(playerUUID, playerStats.deaths);
+      allBowshots.put(playerUUID, playerStats.longestBowKill);
+    }
+
+    Component killMessage = getMessage("stats.kills", sortStats(allKills), ChatColor.GREEN);
+    Component killstreakMessage =
+        getMessage("stats.killstreak", sortStats(allKillstreaks), ChatColor.GREEN);
+    Component deathMessage = getMessage("stats.deaths", sortStats(allDeaths), ChatColor.RED);
+    TopResult bestBowshot = sortStats(allBowshots);
+    String bowMessageKey = (bestBowshot.stat == 1) ? "stats.bowshot.block" : "stats.bowshot.blocks";
+    Component bowshotMessage = getMessage(bowMessageKey, bestBowshot, ChatColor.YELLOW);
+
+    for (MatchPlayer viewer : match.getPlayers()) {
+      viewer.sendMessage(
+          Components.fromLegacyText(
+              ComponentUtils.horizontalLineHeading(
+                  ChatColor.YELLOW
+                      + AllTranslations.get().translate("stats.best", viewer.getBukkit()),
+                  ChatColor.WHITE,
+                  ComponentUtils.MAX_CHAT_WIDTH)));
+
+      viewer.sendMessage(killMessage);
+      viewer.sendMessage(killstreakMessage);
+      viewer.sendMessage(deathMessage);
+      if (bestBowshot.stat != 0) viewer.sendMessage(bowshotMessage);
+    }
+  }
+
+  private class TopResult {
+    UUID uuid;
+    int stat;
+  }
```

Comment:
You don't need this class.

Just use `Map.Entry<UUID, Integer>`

Reply:
This was done because of @TheMolkaPL s suggestion earlier. You can see his reasoning https://github.com/Electroid/PGM/pull/338#discussion_r388542333
I can remove it if you still disagree

Reply:
I would remove it.

Reply:
Sure, i've removed it now.